test/Unit/Language/Lexer/LexerTest.php

```diff
@@ -0,0 +1,942 @@
+            $source,
+            TokenTypes::from(
+                TokenType::INTEGER_BINARY,
+                TokenType::INTEGER_OCTAL,
+                TokenType::INTEGER_HEXADECIMAL,
+                TokenType::INTEGER_DECIMAL,
+                TokenType::SPACE
+            ),
+            new Token(self::range([0, 0], [0, 7]), TokenType::INTEGER_BINARY, '0b100101'),
+            new Token(self::range([0, 8], [0, 8]), TokenType::SPACE, ' '),
+            new Token(self::range([0, 9], [0, 14]), TokenType::INTEGER_DECIMAL, '892837'),
+            new Token(self::range([0, 15], [0, 15]), TokenType::SPACE, ' '),
+            new Token(self::range([0, 16], [0, 23]), TokenType::INTEGER_HEXADECIMAL, '0xFFAAEE'),
+            new Token(self::range([0, 24], [0, 24]), TokenType::SPACE, ' '),
+            new Token(self::range([0, 25], [0, 31]), TokenType::INTEGER_OCTAL, '0o75374'),
+        ];
+
+        yield ($source = '"This is a string literal with \\n escapes \\xB1 \\u5FA9 \\u{1343E}!"') => [
+            $source,
+            TokenTypes::from(
+                TokenType::STRING_LITERAL_DELIMITER,
+                TokenType::STRING_LITERAL_CONTENT,
+                TokenType::ESCAPE_SEQUENCE_SINGLE_CHARACTER,
+                TokenType::ESCAPE_SEQUENCE_HEXADECIMAL,
+                TokenType::ESCAPE_SEQUENCE_UNICODE,
+                TokenType::ESCAPE_SEQUENCE_UNICODE_CODEPOINT
+            ),
+            new Token(self::range([0, 0], [0, 0]), TokenType::STRING_LITERAL_DELIMITER, '"'),
+            new Token(self::range([0, 1], [0, 30]), TokenType::STRING_LITERAL_CONTENT, 'This is a string literal with '),
+            new Token(self::range([0, 31], [0, 32]), TokenType::ESCAPE_SEQUENCE_SINGLE_CHARACTER, '\\n'),
+            new Token(self::range([0, 33], [0, 41]), TokenType::STRING_LITERAL_CONTENT, ' escapes '),
+            new Token(self::range([0, 42], [0, 45]), TokenType::ESCAPE_SEQUENCE_HEXADECIMAL, '\\xB1'),
+            new Token(self::range([0, 46], [0, 46]), TokenType::STRING_LITERAL_CONTENT, ' '),
+            new Token(self::range([0, 47], [0, 52]), TokenType::ESCAPE_SEQUENCE_UNICODE, '\\u5FA9'),
+            new Token(self::range([0, 53], [0, 53]), TokenType::STRING_LITERAL_CONTENT, ' '),
+            new Token(self::range([0, 54], [0, 62]), TokenType::ESCAPE_SEQUENCE_UNICODE_CODEPOINT, '\\u{1343E}'),
+            new Token(self::range([0, 63], [0, 63]), TokenType::STRING_LITERAL_CONTENT, '!'),
+            new Token(self::range([0, 64], [0, 64]), TokenType::STRING_LITERAL_DELIMITER, '"')
+        ];
+
+        $source = <<<AFX
+        """
+            This is "template literal" content with \\n escapes \\xB1 \\u5FA9 \\u{1343E}
+            and embedded expressions: {} :)
+            """
+        AFX;
+        yield $source => [
+            $source,
+            TokenTypes::from(
+                TokenType::TEMPLATE_LITERAL_DELIMITER,
+                TokenType::SPACE,
```

Comment:
interestring. because we allow the space token here its not a `TEMPLATE_LITERAL_CONTENT` (but it probably should be in real world ^^)

Reply:
Yep, the `TemplateLiteralParser` will read this differently :)